### PR TITLE
Lesson 5 Assignments

### PR DESCRIPTION
**Pull request by roslynm**

### CODE REVIEW

File: students/roslyn_m/lesson05/MailroomPart3.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    return
+
+
+# ----------Variables----------#
+person1 = {"Name": 'William Gates', "Total": 653784.49, "Qty": 2, "Avg": 326892.24}
+person2 = {"Name": 'Mark Zuckerberg', "Total": 16396.10, "Qty": 3, "Avg": 5465.37}
+person3 = {"Name": 'Jeff Bezos', "Total": 877.33, "Qty": 1, "Avg": 877.33}
+person4 = {"Name": 'Paul Allen', "Total": 708.42, "Qty": 3, "Avg": 236.14}
+person5 = {"Name": 'John Smith', "Total": 899.98, "Qty": 5, "Avg": 399.21}
+donor_list = [person1, person2, person3, person4, person5]
```
Comment (charlesrobison) [inline students/roslyn_m/lesson05/MailroomPart3.py:117]: I should have caught this last version, but at this point the main structure of your database should be using a dictionary.  This is a list of dictionaries.  You'll want something like:  donor_dict = {'person1': [100.00, 200.50], 'person2: [50.25, 60.00]
etc. so each key is a donor and the values are a list of their donations.

File: students/roslyn_m/lesson05/MailroomPart3.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    """ Create a report of donors to show user.
+                :param: donor_all (list of list)
+                :return: nothing
+                """
+    title = ["Donor Name", "Total Given", "Num Gifts", "Average Gift"]
+    print(f'{title[0]:20}|{title[1]:^12}|{title[2]:^10}|{title[3]:^14}|')
+    print("-" * 60)
+    for person in donor_all:
+        print(f'{person["Name"]:20} ${person["Total"]:^12}{person["Qty"]:9}   ${person["Avg"]:^12}')
+    print("\n")
```
Comment (charlesrobison) [inline students/roslyn_m/lesson05/MailroomPart3.py:83]: Note that new donors aren't being sorted correctly

**Review by charlesrobison — APPROVED** [review]
Mailroom needs some attention - database should be a dictionary and sorting for new donors isn't functioning correctly.  Please take another look and reach out if you have questions.

Ready to merge

**Comment by roslynm** [comment]
Charles,

Thanks for the comments! I have updated my lesson 5 script per your
pickups. I have also submitted lesson 6.

Thanks,
Roslyn

**Pull request merged**